plugins/feedback-backend/src/service/router.ts

```diff
@@ -0,0 +1,298 @@
+  });
+
+  router.get('/:id', async (req, res) => {
+    const feedbackId = req.params.id;
+
+    if (await feedbackDB.checkFeedbackId(feedbackId)) {
+      const feedback: FeedbackModel =
+        await feedbackDB.getFeedbackByUuid(feedbackId);
+      return res.status(200).json({
+        data: feedback,
+        message: 'Feedback fetched successfully',
+      });
+    }
+    return res
+      .status(404)
+      .json({ error: `No feedback found for id ${feedbackId}` });
+  });
+
+  router.get('/:id/ticket', async (req, res) => {
+    const ticketId = req.query.ticketId ? (req.query.ticketId as string) : null;
+    const projectId = req.query.projectId
+      ? (req.query.projectId as string)
+      : null;
+
+    if (ticketId && projectId) {
+      const entityRef: Entity | undefined =
+        await catalogClient.getEntityByRef(projectId);
+      const feedbackType = entityRef?.metadata.annotations?.['feedback/type'];
+      if (feedbackType?.toLowerCase() === 'jira') {
+        let host = entityRef?.metadata.annotations?.['feedback/host'];
+
+        // if host is undefined then
+        // use the first host from config
+        const serviceConfig =
+          config
+            .getConfigArray('feedback.integrations.jira')
+            .find(hostConfig => host === hostConfig.getString('host')) ??
+          config.getConfigArray('feedback.integrations.jira')[0];
+        host = serviceConfig.getString('host');
+        const authToken = serviceConfig.getString('token');
+
+        const resp = await getTicketDetails(host, ticketId, authToken);
+        return res.status(200).json({
+          data: { ...resp },
+          message: 'fetched successfully',
+        });
+      }
+    }
+
+    return res
+      .status(404)
+      .json({ error: `Unable to fetch jira ticket ${ticketId}` });
+  });
+
+  // patch and delete apis
+  router.patch('/:id', async (req, res) => {
+    const feedbackId = req.params.id;
+    const data: FeedbackModel = req.body;
+
+    if (await feedbackDB.checkFeedbackId(feedbackId)) {
+      data.feedbackId = feedbackId;
+      data.updatedAt = new Date().toISOString();
+      const updatedData = await feedbackDB.updateFeedback(data);
+      if (updatedData) {
+        res.status(200).json({
+          data: updatedData,
+          message: 'Feedback updated successfully',
+        });
+        return; // Added return to exit the function after sending the response
+      }
+      res.status(500).json({ error: 'Failed to edit the feedback' });
+      return; // Added return to exit the function after sending the response
+    }
+
+    res.status(404).json({ error: `No feedback found for id ${feedbackId}` });
+  });
+
+  router.delete('/:id', async (req, res) => {
+    const feedbackId = req.params.id;
+    if (await feedbackDB.checkFeedbackId(feedbackId)) {
+      await feedbackDB.deleteFeedbackById(feedbackId);
+      res.status(200).json({ message: 'Deleted successfully' });
```

Comment:
Do you need a return under this statement?

Reply:
yes, thanks, I have added return to the response

Reply:
added return statements after response